### PR TITLE
Ext video apis

### PR DESCRIPTION
**Pull request by tbarbugli**

### CODE REVIEW

**Comment by nuno-vieira** [comment]
Aren't we going to have a separate SDK for video? Including an LLC and UI too?

Ideally, we would have an optional dependency of `StreamChat/Video` which would depend on `StreamVideoUI` and have the video call components already integrated into Chat. Only if the customer wants video calls, would depend on `StreamChat/Video`.

**Comment by jeroenleenarts** [comment]
@martinmitrevski I cherry picked and squashed the work by @tbarbugli.

What needs to happen to get this in a mergable state?

**Comment by martinmitrevski** [comment]
> @martinmitrevski I cherry picked and squashed the work by @tbarbugli.
> 
> What needs to happen to get this in a mergable state?

@jeroenleenarts, I would say first the checks need to pass (tests, etc), also more coverage is needed. Also, a look by @polqf would be more than welcome :)

File: Sources/StreamChat/Repositories/CallRepository.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public class CallRepository {
```
Comment (martinmitrevski) [inline Sources/StreamChat/Repositories/CallRepository.swift:7]: Do we need this public?

**Review by martinmitrevski — COMMENTED** [review]
Looks good 👍 Not sure if the `CallRepository` should be public. Also, we need some more tests and see those failing checks.

**Review by nuno-vieira — APPROVED** [review]

File: Sources/StreamChat/APIClient/Endpoints/Payloads/CallPayloads.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        case agoraUid = "agora_uid"
+        case agoraAppId = "agora_app_id"
+    }
+    
+    /// The call token.
+    let token: String
+    /// The UID related to this token (Agora only).
+    let agoraUid: UInt?
+    /// The Agora App Id (Agora only).
+    let agoraAppId: String?
```
[the remark this answers is not in the document]
Comment (DaemonLoki) [reply Sources/StreamChat/APIClient/Endpoints/Payloads/CallPayloads.swift:19]: We would need to do that indeed. But AFAIK there is nothing planned in that realm. 

I was thinking about refactoring this to have a more convenient `enum` based API, but I think it's an okay solution for these two use cases.

**Comment by DaemonLoki** [comment]
I created a few tests. Not sure if those are sufficient since SonarCloud is still complaining. 
If there's more to test, I'm happy to do so! What do you think @nuno-vieira and @martinmitrevski ?

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**